src/main/java/net/dv8tion/jda/api/managers/GuildScheduledEventManager.java

```diff
@@ -0,0 +1,304 @@
+     *         {@value GuildScheduledEvent#MAX_LOCATION_LENGTH}
+     * @throws java.lang.IllegalStateException
+     *         If the selected {@link GuildScheduledEvent} does not have an end time
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         If the currently logged in account does not have
+     *         {@link net.dv8tion.jda.api.Permission#MANAGE_EVENTS Permission.MANAGE_EVENTS}
+     *
+     * @return GuildScheduledEventManager for chaining convenience
+     *
+     * @see    #setEndTime(TemporalAccessor)
+     * @see    #setLocation(GuildChannel)
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventManager setLocation(@Nonnull String location);
+
+    /**
+     * Sets the time that the selected {@link GuildScheduledEvent} should start at.
+     * Events of {@link GuildScheduledEvent.Type#EXTERNAL Type.EXTERNAL} will automatically
+     * start at this time. Events of {@link GuildScheduledEvent.Type#STAGE_INSTANCE Type.STAGE_INSTANCE}
+     * and {@link GuildScheduledEvent.Type#VOICE Type.VOICE} need to be manually started.
+     * If the {@link GuildScheduledEvent} has not begun after its scheduled start time, it will be automatically cancelled after a few hours.
+     *
+     * @param  startTime
+     *         The time that the selected {@link GuildScheduledEvent} is set to start at.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         <ul>
+     *             <li>If the provided start time is {@code null}</li>
+     *             <li>If the provided start time is before the end time</li>
+     *             <li>If the provided start time is before the current time</li>
+     *         </ul>
+     * @return GuildScheduledEventManager for chaining convenience
+     *
+     * @see    #setEndTime(TemporalAccessor)
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventManager setStartTime(@Nonnull TemporalAccessor startTime);
+
+    /**
+     * Sets the time that the selected {@link GuildScheduledEvent} should end at.
+     * Events of {@link GuildScheduledEvent.Type#EXTERNAL Type.EXTERNAL} will automatically
+     * end at this time, and events of {@link GuildScheduledEvent.Type#STAGE_INSTANCE Type.STAGE_INSTANCE}
+     * and {@link GuildScheduledEvent.Type#VOICE Type.VOICE} will end a few minutes after the last
+     * person has left the channel.
+     *
+     * @param  endTime
+     *         The time that the selected {@link GuildScheduledEvent} is set to end at.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         <ul>
+     *             <li>If the provided end time is before the start time</li>
+     *             <li>If the provided end time is {@code null}</li>
+     *         </ul>
+     * @throws java.lang.IllegalStateException
+     *         If an end time is provided and the event is not set to take place at an external location
```

Comment:
I had made this in order to prevent confusion for devs trying to set an end time when it is not possible for non-external events. There is a check in GuildScheduledManagerImpl but I will remove it now. Discord doesn't care if you provide the end time.